kolibri/core/analytics/management/commands/ping.py

```diff
@@ -70,91 +49,11 @@ def handle(self, *args, **options):
-                if once:
-                    break
-                logger.info("Sleeping for {} minutes.".format(interval))
-                time.sleep(interval * 60)
-                continue
-            except ConnectionError:
-                logger.warn(
-                    "Ping failed (could not connect). Trying again in {} minutes.".format(
-                        checkrate
-                    )
-                )
-            except Timeout:
-                logger.warn(
-                    "Ping failed (connection timed out). Trying again in {} minutes.".format(
-                        checkrate
-                    )
-                )
-            except RequestException as e:
-                logger.warn(
-                    "Ping failed ({})! Trying again in {} minutes.".format(e, checkrate)
-                )
-            if once:
-                break
-            time.sleep(checkrate * 60)
-
-    def perform_ping(self, server):
-
-        url = urljoin(server, "/api/v1/pingback")
-
-        instance, _ = InstanceIDModel.get_or_create_current_instance()
-
-        devicesettings = DeviceSettings.objects.first()
-        language = devicesettings.language_id if devicesettings else ""
-
-        try:
-            timezone = get_current_timezone().zone
-        except Exception:
-            timezone = ""
-
-        data = {
-            "instance_id": instance.id,
-            "version": kolibri.__version__,
-            "mode": conf.OPTIONS["Deployment"]["RUN_MODE"],
-            "platform": instance.platform,
-            "sysversion": instance.sysversion,
-            "database_id": instance.database.id,
-            "system_id": instance.system_id,
-            "node_id": instance.node_id,
-            "language": language,
-            "timezone": timezone,
-            "uptime": int((datetime.now() - self.started).total_seconds() / 60),
-            "timestamp": localtime(),
-            "installer": installation_type(),
-        }
-
-        logger.debug("Pingback data: {}".format(data))
-        jsondata = dump_zipped_json(data)
-        response = requests.post(url, data=jsondata, timeout=60)
-        response.raise_for_status()
-        return json.loads(response.content.decode() or "{}")
-
-    def perform_statistics(self, server, pingback_id):
-        url = urljoin(server, "/api/v1/statistics")
-        channels = [
-            extract_channel_statistics(c) for c in ChannelMetadata.objects.all()
-        ]
-        facilities = [extract_facility_statistics(f) for f in Facility.objects.all()]
-        data = {"pi": pingback_id, "c": channels, "f": facilities}
-        logger.debug("Statistics data: {}".format(data))
-        jsondata = dump_zipped_json(data)
-        response = requests.post(url, data=jsondata, timeout=60)
-        response.raise_for_status()
-        return json.loads(response.content.decode() or "{}")
+                ping_once(started, server)
+            except Exception as e:
```

Comment:
sorry I'm just curious here - when we ping once, we will raise the exception. when we schedule a ping, the exceptions will be caught and only warning logs will be shown. Is this intended? Thank you!!

Reply:
Yes that's the intended behaviour.

Reply:
thanks for clarifying!